### PR TITLE
Improve consistency of handling of Wire.reset() so it can be reused #495

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/YamlTokeniser.java [hunk 1]
```diff
@@ -52,19 +52,22 @@ public class YamlTokeniser {
...
     }
 
     void reset() {
-        freeContexts.addAll(contexts);
         contexts.clear();
+        freeContexts.clear();
         if (temp != null) temp.clear();
-        lineStart = in.readPosition();
+        long pos = in == null ? 0 : in.readPosition();
+        lineStart =
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/wire/YamlTokeniser.java]: is this IntelliJ auto-formatting? Does not look great ;)

File: src/test/java/net/openhft/chronicle/wire/YamlWireTest.java [hunk 2]
```diff
@@ -1040,9 +1036,9 @@ public void testMapReadAndWriteStrings() {
             o.writeEventName(() -> "example")
                     .map(expected);
         });
-       // bytes.readPosition(4);
-       // expectWithSnakeYaml("{example={hello=world, hello1=world1, hello2=world2}}", wire);
-       // bytes.readPosition(0);
+        // bytes.readPosition(4);
+        // expectWithSnakeYaml("{example={hello=world, hello1=world1, hello2=world2}}", wire);
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/wire/YamlWireTest.java:1040]: can we get rid of commented-out code?

File: src/test/java/net/openhft/chronicle/wire/YamlWireTest.java [hunk 3]
```diff
@@ -1365,24 +1358,24 @@ public void testStringMap() {
...
-        // wire = createWire();
-//
-        // Set<String> threeObjects = new HashSet(Arrays.asList(new String[]{"abc", "def", "ghi"}));
-        // wire.write().object(threeObjects);
-//
-        // Set<String> list2 = wire.read()
-        // .object(Set.class);
-        // assertEquals(3, list2.size());
-        // assertEquals("[abc, def, ghi]", list2.toString());
+        // TODO we shouldn't need to create a new wire. should wire.reset()
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/wire/YamlWireTest.java:1361]: given this change, we should do this TODO

**Review by JerryShea — APPROVED** [review]

File: src/test/java/net/openhft/chronicle/wire/YamlWireTest.java (hunk 3 above)
Comment (peter-lawrey) [reply src/test/java/net/openhft/chronicle/wire/YamlWireTest.java:1361]: I did, I need to fix the test now.

Comment (JerryShea) [reply src/test/java/net/openhft/chronicle/wire/YamlWireTest.java:1361]: I tried it locally and it worked

**Review by RobAustin — APPROVED** [review]

**Pull request merged**